File: src/ruby/lib/Definition.rb
class Definition
  
  attr_accessor :headword, :text, :extended_text, :part_of_speech
  
  def initialize(options={})
    @headword = options["headword"]
    @text = options["text"]
    @extended_text = options["extendedText"]
    @part_of_speech = options["partOfSpeech"]
  end
  
end

File: src/objectivec/main/wordnik/Definition.m
//
//  Definition.m
//
//  Created by Tony Tam on 5/17/10.
//  Copyright 2010 Wordnik. All rights reserved.
//

#import "Definition.h"


@implementation Definition
@synthesize text;
@synthesize extendedText;
@synthesize partOfSpeech;

-  (Definition*) init:(NSDictionary*)response {
	self.text = [response objectForKey:@"text"];
	self.extendedText = [response objectForKey:@"extendedText"];
	self.partOfSpeech = [[PartOfSpeech alloc] init:[response objectForKey:@"partOfSpeech"]];

	return self;
}
@end
